test/functional/launch/nested_loop/tests/test-launch-nested-Loop.hpp

```diff
@@ -0,0 +1,194 @@
+// Copyright (c) 2016-23, Lawrence Livermore National Security, LLC
+// and RAJA project contributors. See the RAJA/LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#ifndef __TEST_LAUNCH_NESTED_LOOP_HPP__
+#define __TEST_LAUNCH_NESTED_LOOP_HPP__
+
+#include <numeric>
+
+template <typename INDEX_TYPE, typename WORKING_RES, typename LAUNCH_POLICY,
+          typename THREAD_X_POLICY, typename THREAD_Y_POLICY, typename THREAD_Z_POLICY,
+          typename TEAM_X_POLICY, typename TEAM_Y_POLICY, typename TEAM_Z_POLICY>
+void LaunchNestedLoopTestImpl(INDEX_TYPE M)
+{
+
+  RAJA::TypedRangeSegment<INDEX_TYPE> r1(0, 2*RAJA::stripIndexType(M));
+  RAJA::TypedRangeSegment<INDEX_TYPE> r2(0, 3*RAJA::stripIndexType(M));
+  RAJA::TypedRangeSegment<INDEX_TYPE> r3(0, 4*RAJA::stripIndexType(M));
+
+  RAJA::TypedRangeSegment<INDEX_TYPE> r4(0, 8*RAJA::stripIndexType(M));
+  RAJA::TypedRangeSegment<INDEX_TYPE> r5(0, 2*RAJA::stripIndexType(M));
+  RAJA::TypedRangeSegment<INDEX_TYPE> r6(0, 3*RAJA::stripIndexType(M));
+
+  INDEX_TYPE N1 = static_cast<INDEX_TYPE>(r1.end() - r1.begin());
+  INDEX_TYPE N2 = static_cast<INDEX_TYPE>(r2.end() - r2.begin());
+  INDEX_TYPE N3 = static_cast<INDEX_TYPE>(r3.end() - r3.begin());
+
+  INDEX_TYPE N4 = static_cast<INDEX_TYPE>(r4.end() - r4.begin());
+  INDEX_TYPE N5 = static_cast<INDEX_TYPE>(r5.end() - r5.begin());
+  INDEX_TYPE N6 = static_cast<INDEX_TYPE>(r6.end() - r6.begin());
+
+
+  INDEX_TYPE N = static_cast<INDEX_TYPE>(RAJA::stripIndexType(N1) *
+                                         RAJA::stripIndexType(N2) *
+                                         RAJA::stripIndexType(N3) *
+                                         RAJA::stripIndexType(N4) *
+                                         RAJA::stripIndexType(N5) *
+                                         RAJA::stripIndexType(N6));
+
+  camp::resources::Resource working_res{WORKING_RES::get_default()};
+  INDEX_TYPE* working_array;
+  INDEX_TYPE* check_array;
+  INDEX_TYPE* test_array;
+
+  size_t data_len = RAJA::stripIndexType(N);
+  if ( data_len == 0 ) {
+    data_len = 1;
+  }
+
+  allocateForallTestData<INDEX_TYPE>(data_len,
+                                     working_res,
+                                     &working_array,
+                                     &check_array,
+                                     &test_array);
+
+  //6 threads total
+  constexpr int threads_x = 1;
+  constexpr int threads_y = 2;
+  constexpr int threads_z = 3;
+
+  constexpr int blocks_x = 3;
+  constexpr int blocks_y = 2;
+  constexpr int blocks_z = 1;
+
+  if ( RAJA::stripIndexType(N) > 0 ) {
+
+    std::iota(test_array, test_array + RAJA::stripIndexType(N), 0);
+
+    constexpr int DIM = 6;
+    using layout_t = RAJA::Layout<DIM, INDEX_TYPE,DIM-1>;
+    RAJA::View<INDEX_TYPE, layout_t> Aview(working_array, N6, N5, N4, N3, N2, N1);
+
+    RAJA::launch<LAUNCH_POLICY>
```

Comment:
It looks like this kernel code and zero-length segment kernel code below is repeated in multiple test header files. Would it be better to consolidate in a header file to avoid redundancy?